### DISCUSSION TITLE
help with installing: setup.sh unmet dependancies (clang)

### DISCUSSION THREAD

**Discussion by shott92** [microsoft/AirSim#4776] [Support Q&A]
hi I am hopign someone can help, I have

OS: Ubuntu Jammy Jellyfish 22.04 (Kernel 6.1.5)
CPU Ryzen 7700x
GPU RX 7900 XT -   OpenGL: renderer: AMD Radeon RX 7900 XT (gfx1100 LLVM 15.0.5 DRM 3.49 6.1.5-060105-generic) v: 4.6 Mesa 23.1.0-devel (git-2969850 2023-01-13 jammy-oibaf-ppa)

tried running setup several times, and have been keen to try to resolve myself and install any dependancies, however even using force install and aptitude, I am failing to satisty the setup script:
Here is the output i get when running set-up:
`./setup.sh
+ set -e
+++ dirname ./setup.sh
++ cd .
++ pwd
+ SCRIPT_DIR=/home/xzenner/Programming/AirSim
+ pushd /home/xzenner/Programming/AirSim
+ downloadHighPolySuv=true
+ MIN_CMAKE_VERSION=3.10.0
++ uname
+ '[' Linux == Darwin ']'
+ DEBUG=false
+ [[ 0 -gt 0 ]]
++ uname
+ '[' Linux == Darwin ']'
+ sudo apt-get update
Hit:2 http://archive.ubuntu.com/ubuntu focal InRelease                                                                                            
Hit:3 http://archive.ubuntu.com/ubuntu xenial InRelease                                                                                           
Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease                                                                                            
Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]                                                                           
Ign:7 https://mirror.cov.ukservers.com/linuxmint vera InRelease                                                                                   
Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]                                                                        
Hit:9 https://mirror.cov.ukservers.com/linuxmint vera Release                                                                                     
Get:10 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                                                        
Hit:11 https://ppa.launchpadcontent.net/oibaf/graphics-drivers/ubuntu jammy InRelease                  
Hit:12 https://ppa.launchpadcontent.net/ubuntu-toolchain-r/test/ubuntu jammy InRelease                             
Hit:1 https://apt.llvm.org/xenial llvm-toolchain-xenial-8 InRelease              
Hit:5 https://apt.llvm.org/focal llvm-toolchain-focal InRelease
Fetched 324 kB in 1s (243 kB/s)
Reading package lists... Done
W: http://apt.llvm.org/xenial/dists/llvm-toolchain-xenial-8/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
W: http://apt.llvm.org/focal/dists/llvm-toolchain-focal/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
+ sudo apt-get -y install --no-install-recommends lsb-release rsync software-properties-common wget libvulkan1 vulkan-utils
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
vulkan-utils is already the newest version (1.2.131.1+dfsg1-1).
software-properties-common is already the newest version (2.2.1).
libvulkan1 is already the newest version (1.3.204.1-2).
lsb-release is already the newest version (11.1.0ubuntu4).
wget is already the newest version (1.21.2-2ubuntu1).
rsync is already the newest version (3.2.3-8ubuntu3.1).
0 to upgrade, 0 to newly install, 0 to remove and 2 not to upgrade.
++ lsb_release -rs
++ cut -d. -f1
+ VERSION=21
+ '[' 21 -lt 17 ']'
+ sudo apt-get install -y clang-8 clang++-8 libc++-8-dev libc++abi-8-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note: selecting 'librust-clang-sys-0+gte-clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-0.28+gte-clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'python-clang-8' for regex 'clang++-8'
Note: selecting 'python3-clang-8' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-1.3+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'clang-8-doc' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-0.28.1+gte-clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'clang-8-examples' for regex 'clang++-8'
Note: selecting 'librust-clang-sys+gte-clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'clang-8' for regex 'clang++-8'
Note: selecting 'librust-clang-sys+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-0.28+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-0+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-1+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'libclang-8-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-1.3.0+clang-8-0-dev' for regex 'clang++-8'
Note: selecting 'librust-clang-sys-0.28.1+clang-8-0-dev' for regex 'clang++-8'
Note, selecting 'librust-clang-sys+clang-8-0-dev' instead of 'librust-clang-sys-0+clang-8-0-dev'
Note, selecting 'librust-clang-sys+clang-8-0-dev' instead of 'librust-clang-sys-0.28+clang-8-0-dev'
Note, selecting 'librust-clang-sys+clang-8-0-dev' instead of 'librust-clang-sys-0.28.1+clang-8-0-dev'
Note, selecting 'librust-clang-sys-dev' instead of 'librust-clang-sys-1+clang-8-0-dev'
Note, selecting 'librust-clang-sys-dev' instead of 'librust-clang-sys-1.3+clang-8-0-dev'
Note, selecting 'librust-clang-sys-dev' instead of 'librust-clang-sys-1.3.0+clang-8-0-dev'
libclang-8-dev is already the newest version (1:8.0.1+svn369350-1~exp1~20200112113617.82).
libclang-8-dev set to manually installed.
librust-clang-sys+clang-8-0-dev is already the newest version (0.28.1-6).
python3-clang-8 is already the newest version (1:8.0.1-9).
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies.
 librust-clang-sys+clang-8-0-dev : Depends: librust-clang-sys-dev (= 0.28.1-6) but 1.3.0-1 is to be installed
                                   Depends: librust-clang-sys+gte-clang-3-6-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-3-7-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-3-8-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-3-9-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-4-0-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-5-0-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-6-0-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-7-0-dev (= 0.28.1-6)
                                   Depends: librust-clang-sys+gte-clang-8-0-dev (= 0.28.1-6)
 python-clang-8 : Conflicts: python-clang-x.y
 python3-clang-8 : Conflicts: python-clang-x.y
E: Unable to correct problems, you have held broken packages.`

**Comment by Saddy21** [comment]
Same issue

**Comment by FabianRudolph** [comment]
clang-8 is no longer supported by Ubuntu 22.04.  Only version 9+.
